src/test/java/jp/co/future/uroborosql/mapping/DefaultEntityHandlerTest.java

```diff
@@ -412,6 +414,241 @@ public void testDeleteLockVersionOnly() throws Exception {
 		}
 	}
 
+	@Test
+	public void testBatchInsert() throws Exception {
+
+		try (SqlAgent agent = config.agent()) {
+			agent.required(() -> {
+				TestEntityForInserts test1 = new TestEntityForInserts(1, "name1", 20, LocalDate.of(1990, Month.APRIL, 1),
+						"memo1");
+				TestEntityForInserts test2 = new TestEntityForInserts(2, "name2", 21, LocalDate.of(1990, Month.APRIL, 2),
+						null);
+				TestEntityForInserts test3 = new TestEntityForInserts(3, "name3", 22, LocalDate.of(1990, Month.APRIL, 3),
+						"memo3");
+
+				int count = agent.inserts(Stream.of(test1, test2, test3), InsertsType.BATCH);
+				assertThat(count, is(3));
+
+				TestEntityForInserts data = agent.find(TestEntityForInserts.class, 1).orElse(null);
+				assertThat(data, is(test1));
+				data = agent.find(TestEntityForInserts.class, 2).orElse(null);
+				assertThat(data, is(test2));
+				data = agent.find(TestEntityForInserts.class, 3).orElse(null);
+				assertThat(data, is(test3));
+
+			});
+		}
+	}
+
+	@Test
+	public void testBatchInsert2() throws Exception {
+
+		try (SqlAgent agent = config.agent()) {
+			agent.required(() -> {
+				TestEntityForInserts test1 = new TestEntityForInserts(1, "name1", 20, LocalDate.of(1990, Month.APRIL, 1),
+						"memo1");
+				TestEntityForInserts test2 = new TestEntityForInserts(2, "name2", 21, LocalDate.of(1990, Month.APRIL, 2),
+						null);
+				TestEntityForInserts test3 = new TestEntityForInserts(3, "name3", 22, LocalDate.of(1990, Month.APRIL, 3),
+						"memo3");
+
+				int count = agent.inserts(TestEntityForInserts.class, Stream.of(test1, test2, test3), InsertsType.BATCH);
+				assertThat(count, is(3));
+
+				TestEntityForInserts data = agent.find(TestEntityForInserts.class, 1).orElse(null);
+				assertThat(data, is(test1));
+				data = agent.find(TestEntityForInserts.class, 2).orElse(null);
+				assertThat(data, is(test2));
+				data = agent.find(TestEntityForInserts.class, 3).orElse(null);
+				assertThat(data, is(test3));
+
+			});
+		}
+	}
+
+	@Test
+	public void testBatchInsert3() throws Exception {
+
+		try (SqlAgent agent = config.agent()) {
+			agent.required(() -> {
+				TestEntityForInserts test1 = new TestEntityForInserts(1, "name1", 20, LocalDate.of(1990, Month.APRIL, 1),
+						"memo1");
+				TestEntityForInserts test2 = new TestEntityForInserts(2, "name2", 21, LocalDate.of(1990, Month.APRIL, 2),
+						null);
+				TestEntityForInserts test3 = new TestEntityForInserts(3, "name3", 22, LocalDate.of(1990, Month.APRIL, 3),
+						"memo3");
+
+				int count = agent.inserts(Stream.of(test1, test2, test3), (c, r) -> c > 0,
+						InsertsType.BATCH);
+				assertThat(count, is(3));
+
+				TestEntityForInserts data = agent.find(TestEntityForInserts.class, 1).orElse(null);
+				assertThat(data, is(test1));
+				data = agent.find(TestEntityForInserts.class, 2).orElse(null);
+				assertThat(data, is(test2));
+				data = agent.find(TestEntityForInserts.class, 3).orElse(null);
+				assertThat(data, is(test3));
+
+			});
+		}
+	}
+
```

Comment:
Please add a test case of batchInsert.
- Case where batch insert in loop and batch insert after loop are executed
  ex) 4rows and condition (c, r) => c == 3

Reply:
I added test cases!

Reply:
Thanks